utils/queuer.go

```diff
@@ -99,17 +102,76 @@ func (q GeneralKVQueue) Keys() []Key {
 	return keys
 }
 
-// ImportState should only be used to populate state at genesis. Panics if the given state is invalid
-func (q GeneralKVQueue) ImportState(state map[string]codec.ProtoMarshaler, validator ...func(map[string]codec.ProtoMarshaler) error) {
+// ExportState exports the given queue's state from the kv store
+func (q GeneralKVQueue) ExportState() (state QueueState) {
+	state.Items = make(map[string]QueueState_Item)
+
+	iter := sdk.KVStorePrefixIterator(q.store.KVStore, q.name.AsKey())
+	defer CloseLogError(iter, q.logger)
+
+	for ; iter.Valid(); iter.Next() {
+		var key gogoprototypes.BytesValue
+		q.store.cdc.MustUnmarshalLengthPrefixed(iter.Value(), &key)
+
+		item := QueueState_Item{
+			Key:   key.Value,
+			Value: q.store.GetRaw(KeyFromBz(key.Value)),
+		}
+
+		state.Items[string(iter.Key())] = item
+	}
+
+	return state
+}
+
+// ImportState imports the given queue state into the kv store
+func (q GeneralKVQueue) ImportState(state QueueState, validator ...func(QueueState) error) {
```

Comment:
```suggestion
func (q GeneralKVQueue) ImportState(state QueueState, validate ...func(QueueState) error) {
```

Reply:
on a related note, we do you pass in the state and the validate function instead of passing in validated state?